### PR TITLE
Add error messages for DSC configuration in summary page

### PR DESCRIPTION
**Pull request by AmelBawa-msft**
## Summary of the pull request
- Copy DSC error messages over from winget repository:
  - [ErrorCodes.cs](https://github.com/microsoft/winget-cli/blob/master/src/PowerShell/Microsoft.WinGet.Configuration.Engine/Exceptions/ErrorCodes.cs)
  - [Resources.resx](https://github.com/microsoft/winget-cli/blob/master/src/PowerShell/Microsoft.WinGet.Configuration.Engine/Resources/Resources.resx)

## Validation steps performed
<table>
<tr><th>Before</th></tr>
<tr>
<td>

</td>
</tr>
<tr><th>After</th></tr>
<tr>
<td>

</td>
</tr>
</table>

## PR checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: tools/SetupFlow/DevHome.SetupFlow/Strings/en-us/Resources.resw [hunk 1]
```diff
@@ -1263,4 +1255,97 @@
     <value>Add another account</value>
     <comment>Menu flyout item content to add another account</comment>
   </data>
+  <data name="ConfigurationFieldInvalidType" xml:space="preserve">
```
Comment (krschau) [inline tools/SetupFlow/DevHome.SetupFlow/Strings/en-us/Resources.resw:1258]: I assume it's not an issue for us to be re(?)-localizing errors from somewhere else?

**Review by krschau — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow/Strings/en-us/Resources.resw (hunk 1 above)
Comment (AmelBawa-msft) [reply tools/SetupFlow/DevHome.SetupFlow/Strings/en-us/Resources.resw:1258]: Correct. Only the HRESULT code is communicate over from the API. Error localized messages are up  to the user (in this case Dev Home) to handle.

**Review by vineeththomasalex — APPROVED** [review]

File: tools/SetupFlow/DevHome.SetupFlow/Models/ConfigurationUnitResult.cs [hunk 2]
```diff
@@ -19,6 +19,8 @@ public ConfigurationUnitResult(ApplyConfigurationUnitResult result)
         Intent = result.Unit.Intent.ToString();
         IsSkipped = result.State == ConfigurationUnitState.Skipped;
         HResult = result.ResultInformation?.ResultCode?.HResult ?? HRESULT.S_OK;
+        ResultSource = result.ResultInformation?.ResultSource ?? ConfigurationUnitResultSource.None;
+        Details = result.ResultInformation.Details;
```
Comment (vineeththomasalex) [inline tools/SetupFlow/DevHome.SetupFlow/Models/ConfigurationUnitResult.cs:23]: Nit: ResultInformation doesn't need a null-conditional here?

Comment (AmelBawa-msft) [inline tools/SetupFlow/DevHome.SetupFlow/Models/ConfigurationUnitResult.cs:23]:
```suggestion
        Details = result.ResultInformation?.Details;
```

**Pull request merged**